### PR TITLE
RELATED: RAIL-4101 add support for render mode in ui dashboard state

### PR DESCRIPTION
**Pull request by MilanLempera**
---

Supported PR commands:

| Command                  | Description             |
| ------------------------ | ----------------------- |
| `ok to test`             | Re-run standard checks  |
| `extended test`          | BackstopJS tests        |
| `extended check sonar`   | SonarQube tests         |
| `extended check cypress` | Cypress E2E tests       |
| `extended check plugins` | Dashboard plugins tests |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `check-extended-cypress` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7575/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9431/ : FAILURE in 5m 02s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7579/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9434/ : SUCCESS in 11m 21s

File: libs/sdk-ui-dashboard/src/model/store/ui/uiSelectors.ts [hunk 1]
```diff
@@ -123,3 +123,13 @@ export const selectMenuButtonItemsVisibility = createSelector(
     selectSelf,
     (state) => state.menuButton.itemsVisibility ?? {},
 );
+
+/**
+ * @alpha
+ */
+export const selectIsInViewMode = createSelector(selectSelf, (state) => state.renderMode === "view");
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/store/ui/uiSelectors.ts:130]: Maybe also a plain `selectRenderMode` could also be useful in addition to these two?

File: libs/sdk-ui-dashboard/src/presentation/dashboard/types.ts [hunk 2]
```diff
@@ -460,6 +461,14 @@ export interface IDashboardBaseProps {
...
      */
     permissions?: IWorkspacePermissions;
+
+    /**
+     * Specify which render mode will be used for initial rendering.
+     *
+     * @remarks
+     * If you do not specify initialRenderMode, the dashboard component will be display in view mode.
+     */
+    initialRenderMode?: RenderMode;
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/dashboard/types.ts:471]: We should probably mark most of these new edit mode props and interfaces as `@internal` before edit mode GA, so if there will be some SDK relase meanwhile, it won't be public yet. @no23reason @lupko what do you think?

Comment (kandl) [inline libs/sdk-ui-dashboard/src/presentation/dashboard/types.ts:471]: We should probably mark most of these new edit mode props and interfaces as `@internal` before edit mode GA, so if there will be some SDK relase meanwhile, it won't be public yet. @no23reason @lupko what do you think?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7614/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9475/ : FAILURE in 5m 24s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7615/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9476/ : FAILURE in 7m 53s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7616/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9477/ : FAILURE in 5m 05s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7617/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9478/ : FAILURE in 7m 34s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7618/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9479/ : FAILURE in 10m 26s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7619/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9480/ : SUCCESS in 9m 52s

File: libs/sdk-ui-dashboard/src/model/events/ui.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        ctx,
+        correlationId,
+        payload: {
+            renderMode,
+        },
+    };
+}
+
+/**
+ * Tests whether the provided object is an instance of {@link DashboardWidgetExecutionStarted}.
```
Comment (kandl) [inline libs/sdk-ui-dashboard/src/model/events/ui.ts:48]: Link is not correct (should be `DashboardRenderModeChanged`, not `DashboardWidgetExecutionStarted`)

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/7622/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/9483/ : SUCCESS in 10m 28s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR2440

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Pull request merged**